src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2018,6 +2018,214 @@ TEST_F(
   EXPECT_EQ(1u, app_list.size());
 }
 
+TEST_F(
+    ApplicationManagerImplTest,
+    RegisterApplication_WEPNonMediaAppRegisterActivateDeactivateExit_EXPECT_CORRECT_HMI_STATES) {
+  EXPECT_CALL(
+      mock_session_observer_,
+      GetDataOnSessionKey(kConnectionKey,
+                          _,
+                          _,
+                          testing::An<connection_handler::DeviceHandle*>()))
+      .WillOnce(DoAll(SetArgPointee<3u>(kDeviceId), Return(0)));
+
+  smart_objects::SmartObject rai(smart_objects::SmartType_Map);
+  smart_objects::SmartObject& params = rai[strings::msg_params];
+  params[strings::app_id] = kAppId;
+  params[strings::language_desired] = mobile_api::Language::EN_US;
+  params[strings::hmi_display_language_desired] = mobile_api::Language::EN_US;
+
+  rai[strings::params][strings::connection_key] = kConnectionKey;
+  rai[strings::msg_params][strings::app_name] = kAppName;
+  rai[strings::msg_params][strings::app_hmi_type] = "WEB_VIEW";
+  rai[strings::msg_params][strings::is_media_application] = false;
+  rai[strings::msg_params][strings::sync_msg_version][strings::minor_version] =
+      APIVersion::kAPIV2;
+  rai[strings::msg_params][strings::sync_msg_version][strings::major_version] =
+      APIVersion::kAPIV6;
+
+  rai[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
```

Comment:
@AByzhynar I believe it's a good candidate for a separate function in the `ApplicationManagerImplTest` like `GenerateWEPRaiRequest` which returns you a generated smart object. You may pass some specific values as input params

Reply:
@AKalinich-Luxoft Done in e570bb7a3d5d48b9e64a5172d70804c532ec880f
Thank you